src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -8866,6 +9705,77 @@ class C2 : I
             comp.VerifyEmitDiagnostics();
         }
 
+        [Fact]
+        public void OverridesAndInterfaceImplementations_Out_04()
+        {
+            var source =
+@"#nullable enable
+abstract class A
+{
+    public abstract void F1<T>(out T t);
+    public abstract void F2<T>(out T? t);
+}
+class B1 : A
+{
+    public override void F1<T>(out T t) { t = default!; }
+    public override void F2<T>(out T? t) where T : default { t = default; }
```

Comment:
What is this actually testing wrt ref fields? Ref fields can only be in ref structs, and ref structs cannot be used as generic type parameters.

Reply:
> What is this actually testing wrt ref fields?

It's not testing ref fields. I believe I added this test when there unexpected errors around overrides, `T?`, and `scoped` (since the `out` parameters are implicitly `scoped`). I think the test will be more meaningful after addressing the issue link below and adding `[UnscopedRef]` cases.